weaver/wps_restapi/jobs/utils.py

```diff
@@ -433,51 +460,159 @@ def get_results(  # pylint: disable=R1260
-        - Parameter ``response: raw|document``
-        - Parameter ``transmissionMode: value|reference`` per output if ``response: raw``.
+        - Body parameter ``outputs`` with the amount of *requested outputs* to be returned.
+        - Body parameter ``response: raw|document`` for content representation.
+        - Body parameter ``transmissionMode: value|reference`` per output.
+        - Header parameter ``Prefer: return=representation|minimal`` for content representation.
+        - Overrides, for any of the previous parameters, allowing request of an alternate representation.
+
+    Resolution order/priority:
+
+    1. :paramref:`override_contents`
+    2. :paramref:`override_headers`
+    3. :paramref:`job` definitions
+
+    The logic of the resolution order is that any body parameters resolving to an equivalent information provided
+    by header parameters will be more important, since ``Prefer`` are *soft* requirements, whereas body parameters
+    are *hard* requirements. The parameters stored in the :paramref:`job` are defined during :term:`Job` submission,
+    which become the "default" results representation if requested as is. If further parameters are provided to
+    override during the results request, they modify the "default" results representation. In this case, an header
+    provided in the results request overrides the body parameters from the original :term:`Job`, since their results
+    request context is "closer" than the ones at the time of the :term:`Job` submission.
 
     .. seealso::
         More details available for each combination:
         - https://docs.ogc.org/is/18-062r2/18-062r2.html#sc_execute_response
         - https://docs.ogc.org/is/18-062r2/18-062r2.html#_response_7
+        - :ref:`proc_op_job_results`
+        - :ref:`proc_exec_results`
 
-    :param job: Job for which to generate the results response.
+    :param job: Job for which to generate the results response, which contains the originally submitted parameters.
     :param container: Application settings.
-    :param headers: Additional headers to provide in the response.
+    :param request_headers: Original headers submitted to the request that leads to this response.
+    :param response_headers: Additional headers to provide in the response.
+    :param results_headers: Headers that override originally submitted job parameters when requesting results.
+    :param results_contents: Body contents that override originally submitted job parameters when requesting results.
     """
     raise_job_dismissed(job, container)
     raise_job_bad_status(job, container)
+    settings = get_settings(container)
+
+    # FIXME: apply converters (https://github.com/crim-ca/weaver/pull/548)
+    #       - test_execute_single_output_response_document_alt_format_json
+    #       - test_execute_single_output_response_document_alt_format_yaml
+    #       - test_execute_single_output_multipart_accept_alt_format
```

Comment:
Shouldn't the comment be in `get_job_results_single` and `get_job_results_multipart` where the conversion will be applied ?

Reply:
Yes. Will remove them. Similar FIXME are mentioned directly on those functions instead.